File: src/main/java/play/pluv/playlist/application/dto/PlayListOverViewResponse.java
package play.pluv.playlist.application.dto;

import java.util.List;
import lombok.Builder;
import play.pluv.playlist.domain.PlayList;
import play.pluv.playlist.domain.PlayListId;

@Builder
public record PlayListOverViewResponse(
    String id,
    String thumbNailUrl,
    Integer songCount,
    String name,
    String source
) {

  public static List<PlayListOverViewResponse> createList(final List<PlayList> playLists) {
    return playLists.stream()
        .map(PlayListOverViewResponse::from)
        .toList();
  }

  public static PlayListOverViewResponse from(final PlayList playList) {
    final PlayListId playListId = playList.getPlayListId();
    return new PlayListOverViewResponseBuilder()
        .thumbNailUrl(playList.getThumbNailUrl())
        .songCount(playList.getSongCount())
        .name(playList.getName())
        .source(playListId.musicStreaming().getName())
        .id(playListId.id())
        .build();
  }
}

File: src/main/kotlin/play/pluv/playlist/application/dto/PlayListOverViewResponse.kt
package play.pluv.playlist.application.dto

import play.pluv.playlist.domain.PlayList


data class PlayListOverViewResponse(
    val id: String,
    val thumbNailUrl: String,
    val songCount: Int?,
    val name: String,
    val source: String
) {
    companion object {
        fun createList(playLists: List<PlayList>): List<PlayListOverViewResponse> {
            return playLists.map(::from)
        }

        fun from(playList: PlayList): PlayListOverViewResponse {
            val playListId = playList.playListId
            return PlayListOverViewResponse(
                id = playListId.id,
                thumbNailUrl = playList.thumbNailUrl,
                songCount = playList.songCount,
                name = playList.name,
                source = playListId.musicStreaming.name,
            )
        }
    }
}
